MERN/Topic/Frontend/React/Map_in_JSX.md

```diff
@@ -0,0 +1,69 @@
+# **Using .map() in JSX**
+---
+## **What is map?**
+<br>
+
+A Map is a data collection type that holds **key-value pairs** and remembers the original insertion order of the keys. Any value (both objects and primitive values) may be used as either a key or a value. We cannot store a duplicate pair in the map(). It is because of the uniqueness of each stored key. 
+
+**Use:** It is mainly used for fast searching and looking up data.
+
+**In React**, a Map function is a Javascript function for handling arrays. Many times we come upon a situation where we are repetitively passing reusable components that come from an array. One way to iterate over it is to use a for loop or a forEach() function. However, the only drawback is that it mutates the array. The better approach in such type of situation is to use mapping component which can easily map all the individual custom pieces of data to each of the custom components without mutating the array and are easier to debug.

```

Comment:
Please try to provide little explanation or example about how they mutate and map doesn't.